### DISCUSSION TITLE
Please enable trivy as a pre-commit hook.

### DISCUSSION THREAD

**Discussion by MichaelMorozovCodit** [aquasecurity/trivy#6368] [Ideas]
### Description

Please add trivy as a pre-commit feature. The requirements, per [pre-commit](https://[pre-commit](https://pre-commit.com/#new-hooks).com/#new-hooks) are:

> pre-commit currently supports hooks written in [many languages](https://pre-commit.com/#supported-languages). As long as your git repo is an installable package (gem, npm, pypi, etc.) or exposes an executable, it can be used with pre-commit. Each git repo can support as many languages/hooks as you want.
 The hook must exit nonzero on failure or modify files.
 A git repo containing pre-commit plugins must contain a .pre-commit-hooks.yaml file [...]

Currently added a pre-commit feature manually, but this doesn't prevent me from committing vulnerabilites.

### Target

Git Repository

### Scanner

None

**Comment by DmitriyLewen** [comment]
Hello @MichaelMorozovCodit 
Thanks for your interest to Trivy.

Take a look this - https://aquasecurity.github.io/trivy/v0.50/ecosystem/ide/#trivy-pre-commit-community

Regards, Dmitriy

**Reply by MichaelMorozovCodit** [reply]
Hi Dmitriy

Thanks for your reply. I've seen this one, and unfortunately it's dependent on v0.49.1, needs docker, and only supports the `trivyconfig-docker` and `trivyfs-docker` command. Would be amazing to have a dedicated pre-commit hook with trivy's full support, flexibility and functionality. 

Kind regards,

Michael

**Reply by DmitriyLewen** [reply]
As we wrote in https://github.com/aquasecurity/trivy/pull/2490#issuecomment-1313682146 and https://github.com/aquasecurity/trivy/pull/2490#issuecomment-1483879768, we can't add all proposed options.

> it's dependent on v0.49.

It looks like `mxab` updates its repository regularly.

> needs docker, and only supports the trivyconfig-docker and trivyfs-docker command.

I think you should suggest changes to https://github.com/mxab/pre-commit-trivy. I'm sure you can easily add the necessary options.

**Comment by MichaelMorozovCodit** [comment]
Understood, thanks for taking the time to reply!
